### PR TITLE
[AD-751] add null checks and remove isNullOrWhitespace checks in getColumns and getTables

### PR DESCRIPTION
**Pull request by alinaliBQ**
### Summary

replace isNullOrWhitespace checks with null checks in appropriate areas for getColumns and getTables.

### Description

- [x] update catalog and schemaPattern to have null checks instead of isNullOrWhitespace check in getColumns
- [x] update catalog and schemaPattern to have null checks instead of isNullOrWhitespace check in getTables
- reason: make getColumns / getTables functions comply with the Orcale documentation. [Reference](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#getTables(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String%5B%5D))
### Related Issue

https://bitquill.atlassian.net/browse/AD-751

### Additional Reviewers
@affonsoBQ
@alinaliBQ
@andiem-bq
@alexey-temnikov
@birschick-bq
@mitchell-elholm 
@RoyZhang2022

### CODE REVIEW

**Review by affonsov — APPROVED** [review]

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java [hunk 1]
```diff
@@ -224,7 +224,7 @@ public ResultSet getTables(final String catalog, final String schemaPattern,
         if (isNullOrWhitespace(catalog)
                 && (types == null || types.length == 0 || Arrays.stream(types)
                         .anyMatch(s -> isNullOrWhitespace(s) || s.equals("TABLE")))) {
-            if (isNullOrWhitespace(schemaPattern)
+            if (schemaPattern == null
                     || properties.getDatabase().matches(convertPatternToRegex(schemaPattern))) {
                 addTablesForSchema(tableNamePattern, metaData);
```
Comment (RoyZhang2022) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java:229]: The tableNamePattern in the addTablesForSchema seems also need this change.

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ public ResultSet getColumns(final String catalog, final String schemaPattern,
             final String tableNamePattern, final String columnNamePattern) throws SQLException {
         final List<List<Object>> metaData = new ArrayList<>();
         if (isNullOrWhitespace(catalog)) {
-            if (isNullOrWhitespace(schemaPattern)
+            if (schemaPattern == null
```
Comment (RoyZhang2022) [inline src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java:310]: The columnNamePattern in addTablesForSchema also needs this change?

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java [hunk 3]
```diff
@@ -186,6 +186,17 @@ void testGetTables() throws SQLException {
         }
     }
 
+    /**
+     * Test getTables returns empty ReseultSet when empty string schemaPattern is passed.
```
Comment (RoyZhang2022) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: typo

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java (hunk 1 above)
Comment (alinaliBQ) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java:229]: On the oracle documentation, it says that 
`tableNamePattern - a table name pattern; must match the table name as it is stored in the database.` So I think that means tableNamePattern cannot be null. `isNullOrWhitespace(tableNamePattern)` needs to be removed in `addTablesForSchema`. I will double check with our own JDBC documentation

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java (hunk 3 above)
Comment (alinaliBQ) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: could you let me know where is the typo please?

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java (hunk 1 above)
Comment (alinaliBQ) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java:229]: yea our own JDBC limitation doc does not say anything about tableNamePattern, so I'll remove the `isNullOrWhitespace(tableNamePattern)` check. I'll do the same regarding your other comment on `columnNamePattern` in `addColumnsForSchema`

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java (hunk 3 above)
Comment (RoyZhang2022) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: ReseultSet

File: src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java (hunk 2 above)
Comment (alinaliBQ) [reply src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java:310]: I have made the change

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java (hunk 3 above)
Comment (alinaliBQ) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java]: I've fixed the typos.

**Comment by affonsov** [comment]
Can you double-check the coverage I see that the branches and code coverage lowered a bit after the test changes.

**Comment by alinaliBQ** [comment]
> Can you double-check the coverage I see that the branches and code coverage lowered a bit after the test changes.

@affonsoBQ  I have added more test cases. It should increase code coverage after the new code badges are generated

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java [hunk 4]
```diff
@@ -186,6 +190,26 @@ void testGetTables() throws SQLException {
...
+    void testGetMetadataTablesEmpty() throws SQLException {
+        // Catalog pattern, schema pattern, table name pattern, table types
+        final String[][] tests = new String [][]{
+                {null, "", COLLECTION_BASIC, "TABLE"},
+                {" ", "", COLLECTION_BASIC, "TABLE"},
+                {"non-existing catalog", DATABASE, COLLECTION_BASIC, "non-existing table"},
+                {" ", "", COLLECTION_BASIC},
+        };
+        for (String[] test : tests) {
+            final String[] tableTypes = test.length == 4 ? new String[]{test[3]} : null;
```
Comment (RoyZhang2022) [inline src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java:208]: Seems test[0], test[1] and test[2] are not useful here.

**Comment by alinaliBQ** [comment]
ok it seems that the code coverage still needs more improvement. It is now better than before my last commit but still not as high as before this PR. I'll work on this

File: src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java (hunk 4 above)
Comment (alinaliBQ) [reply src/test/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaDataTest.java:208]: good catch! I forgot to replace `null, "", COLLECTION_BASIC` with `test[0], test[1] and test[2]`

**Review by RoyZhang2022 — APPROVED** [review]

**Review by birschick-bq — APPROVED** [review]

**Pull request merged**